types/types.go

```diff
@@ -175,16 +180,15 @@ func (m *memMarker) Status(alert model.Fingerprint) AlertStatus {
 	s, found := m.m[alert]
 	if !found {
 		s = &AlertStatus{}
-		m.m[alert] = s
```

Comment:
Is it ok to get rid of populating marker with empty state in the `Status()` method? This way we avoid lock switching and as soon as we can `Set*()` we'll insert proper status into the map.